tsdb/chunkenc/histo.go

```diff
@@ -377,11 +394,14 @@ func (a *HistoAppender) AppendHistogram(t int64, h histogram.SparseHistogram) {
 			putVarint(a.b, a.buf64, buck)
 		}
 	case 1:
-
 		tDelta = t - a.t
 		cntDelta = int64(h.Count) - int64(a.cnt)
 		zcntDelta = int64(h.ZeroCount) - int64(a.zcnt)
 
+		if value.IsStaleNaN(h.Sum) {
+			tDelta, cntDelta, zcntDelta = 0, 0, 0
```

Comment:
I think we still want to store the timestamp normally for a stale marker. If that's the case, `tDelta` should be treated normally and not set to 0 here.

Reply:
Good catch! But I am confused why are the tests not failing then. Let me find out...

Reply:
I get it now - the dod is 0 since timestamp are increasing at the same rate, hence not captured by tests.